src/components/Chat/ChatContainer.js

```diff
@@ -70,41 +68,50 @@ class ChatContainer extends Component {
       stage: chatDetails.stage ? chatDetails.stage : null,
       featurePermissions: chatDetails.featurePermissions ? chatDetails.featurePermissions : null,
       apiGatewayEndpoint: chatDetails.apiGatewayEndpoint ? chatDetails.apiGatewayEndpoint : null,
-    }
+    };
     this.logger && this.logger.info("Chat session meta data:", logContent);
     this.submitChatInitiation(chatDetails, success, failure);
   }
 
   /**
    * Initiate a chat in 2 steps.
-   * 
+   *
    * Step 1: Create a chat session within Amazon Connect (more details in ChatInitiator.js)
    * This step provides us with a 'chatDetails' object that contains among others:
    * - Auth Token
    * - Websocket endpoint
    * - ContactId
    * - ConnectionId
-   * 
+   *
    * Step 2: Connect to created chat session.
    * Open a websocket connection via Chat.JS (more details in ChatSession.js)
-   * 
-   * @param {*} input 
-   * @param {*} success 
-   * @param {*} failure 
+   *
+   * @param {*} input
+   * @param {*} success
+   * @param {*} failure
    */
   async submitChatInitiation(input, success, failure) {
     this.setState({ status: "Initiating" });
 
     try {
       const chatDetails = await initiateChat(input);
       const chatSession = await this.openChatSession(chatDetails, input.name, input.region, input.stage);
+      const attachmentsEnabled =
+        (input.featurePermissions && input.featurePermissions[CHAT_FEATURE_TYPES.ATTACHMENTS]) ||
+        (chatDetails.featurePermissions && chatDetails.featurePermissions[CHAT_FEATURE_TYPES.ATTACHMENTS]);
+      const richMessagingFeatureEnabled =
+        !!(input.featurePermissions && input.featurePermissions.hasOwnProperty(CHAT_FEATURE_TYPES.MESSAGING_MARKDOWN)) &&
+        input.featurePermissions[CHAT_FEATURE_TYPES.MESSAGING_MARKDOWN];
+      const markdownMessageTypeSupported =
+        typeof input.supportedMessagingContentTypes === "string" ? input.supportedMessagingContentTypes.split(",").includes(ContentType.MESSAGE_CONTENT_TYPE.TEXT_MARKDOWN) : false;
 
       this.setState({
         status: "Initiated",
         chatSession: chatSession,
         composerConfig: {
-          attachmentsEnabled: (input.featurePermissions && input.featurePermissions[CHAT_FEATURE_TYPES.ATTACHMENTS]) || (chatDetails.featurePermissions && chatDetails.featurePermissions[CHAT_FEATURE_TYPES.ATTACHMENTS])
-        }
+          attachmentsEnabled,
+          richMessagingEnabled: markdownMessageTypeSupported && richMessagingFeatureEnabled,
```

Comment:
same comment - I don't think we want to ask customers to enable the feature in 2 places